### ISSUE TITLE
Making Discussion Channel to favourite

### ISSUE THREAD

**Issue by ABHAYJAIN2601** [RocketChat/Rocket.Chat#21041]
### Description:

If user try  to add discussion channel to favourite It don't show any pop-up/message if user is not the part of that Discussion channel

### Steps to reproduce:

1. Try to add any Discussion channel to your favourite (that you are not part of it)

### Expected behavior:

At-least we can show any can of pop-up/message

### Actual behavior:

it's like kind of error why i can not mark it favourite

https://user-images.githubusercontent.com/51324264/110456821-02a7f480-80f0-11eb-8e23-4f2fc9cd9eec.mp4

### Server Setup Information:

- Version of Rocket.Chat Server:  3.13.0-develop
- Operating System: linux
- Deployment Method: 
- Number of Running Instances: 
- DB Replicaset Oplog: Enabled
- NodeJS Version: 12.18.4 - x64
- MongoDB Version:4.2.8

### Client Setup Information

- Desktop App or Browser Version:Chorme
- Operating System:Window 10

**Comment by ABHAYJAIN2601** [comment]
I started looking into this issue. If you have any other idea please share it here!

**Comment by ABHAYJAIN2601** [comment]
@dougfabris  Please review this !

**Comment by ABHAYJAIN2601** [comment]
Yeah! but if user get pop-up message to show it will look more better right?

**Comment by himani-mehra** [comment]
Hi @ABHAYJAIN2601  , I would love to take on this issue and contribute to the project. Could you please assign it to me if it's available? Looking forward to your response. Thanks!
